var/spack/repos/builtin/packages/improved-rdock/package.py

```diff
@@ -82,12 +92,12 @@ def test(self):
             "-s",
             "1",
         ]
-        self.run_test(str(mpiexe), options=opts)
-
-        opts = [join_path(self.test_suite.current_test_data_dir, "test.sh")]
-        self.run_test("bash", options=opts)
+        self.run_rdock_test(str(self.spec["mpi"].prefix.bin.mpirun), opts)
 
-        pythonexe = self.spec["python"].command.path
+    def test_pythonexe(self):
+        """Check pythonexe with output"""
+        exe = which(self.spec["python"].command.path)
```

Comment:
`which()` is unnecessary with `command`:

```suggestion
        exe =self.spec["python"].command
```